src/signals/incident/services/object-flatten/index.js

```diff
@@ -0,0 +1,10 @@
+// based on source: https://30secondsofcode.org/object#flattenobject
+const flattenObject = (obj, prefix = '', separator = '.') =>
+  Object.keys(obj).reduce((acc, k) => {
+    const pre = prefix.length ? prefix + separator : '';
+    if (typeof obj[k] === 'object') Object.assign(acc, flattenObject(obj[k], pre + k, separator));
```

Comment:
Same comment about type checking applies here.
Also, use template literals

Reply:
Disagree, highly specialized function. No need to go overboard with type checking and the likes.